### ISSUE TITLE
Info about the project

### ISSUE THREAD

**Issue by trane77** [misfotto/enel-energia-api#1]
Hello,

I've just found your repository, so I'd be glad to know more information about it if possible. 

Looks like no code has been uploaded, is it just an initial idea, or it is abandoned? 

In any case, I really need to get more information about how to connect to enel APIs, if you have something for me, that would be great. 

Hope to hear something from you soon,

Cheers mate 

Federico

**Comment by misfotto** [comment]
Hello,

this project as asual happens is stored in some harddrive somewhere in my house.
Never pushed, my fault! :D

The project uses the gecko driver to crawl, parse and take actions on the actual enel website.
It's not completed but has started taking shape.

Do you want to partecipate in the development? i will push the code as soon as i found it :D
NOTE: the api will stop working as intended when the enel website changes something major in the frontend and/or in the url shema until the crawling engine will be updated.

**Comment by trane77** [comment]
Hi, thanks for fast reply. 

No worries about the code, my concern is about understanding if we are talking about the same company. 
For enel I intend the italian energy provider (https://www.enel.it/).

Then if we are talking about the same company, well I'm interested into finding their APIs or any documentation regarding how to connect with them and get information about a contract by passing some data (contract Id?).

What you've written in the NOTE makes me think that we are talking about the same project, I've already heard about those changes in the url schema. 

I can contribute if we're talking about the same company!

edit: If you're italian, we can have a quick call later, I'll give you my mobile number

**Comment by misfotto** [comment]
Yes we are talking about the same company, and yes, i'm italian :)

Enel does not provide any api so my api is kinda like a bot that uses the website extracting only the interesting data.

**Comment by trane77** [comment]
Okay I See..

Praticamente dovrei realizzare un servizio che consenta di caricare i contratti, senza utilizzare un tool che enel mette già a disposizione dei propri partner/rivenditori. 

So che esiste un software in grado di svolgere questa funzione, ma è soggetto al problema del cambio URL, proprio come mi dicevi a proposito del bot.

Ora non so di che cosa si occupi il tuo servizio, ma secondo te è fattibile quello che sto cercando di fare? 

Grazie 

PS: se preferisci continuo a scrivere in inglese, ma a dire il vero, ogni tanto fa piacere parlare con la propria lingua madre qui ;)

**Comment by shiner66** [comment]
Ciao!
Sto cercando un progetto simile che mi permetta di prelevare i dati sui consumi dal sito enel energia, sei riuscito a trovare il codice? ;)

**Comment by trane77** [comment]
Per il momento le ricerche si sono fermate, ho inviato una mail ad enel, ma è praticamente impossibile avere una risposta. Nel tuo caso però è diverso, il sito è online, dovresti "solo" eseguire l'accesso e replicare le chiamate alle api utilizzando (immagino) il token di sessione, o simile.

**Comment by shiner66** [comment]
> Per il momento le ricerche si sono fermate, ho inviato una mail ad enel, ma è praticamente impossibile avere una risposta. Nel tuo caso però è diverso, il sito è online, dovresti "solo" eseguire l'accesso e replicare le chiamate alle api utilizzando (immagino) il token di sessione, o simile. 

Ciao!
In realtà dopo aver scavato un pochino mi sono reso conto che la mia idea è abbastanza infattibile, in quanto avrei bisogno di statistiche **almeno** giornaliere, mentre enel le mette a disposizione il mese dopo, quindi poco utili ai miei scopi :(

**Comment by trane77** [comment]
Ho capito... con una ricerca veloce ho trovato [questo documento](http://www.cet.to.it/index.php/news/45-portale-consumi-per-accedere-ai-dati-relativi-alle-forniture-di-ee-e-gas-naturale), che rimanda al [portale consumi](https://www.consumienergia.it/portaleConsumi/), sembra interessante, non so se il progetto sia attivo (sembra di sì), ma dovrebbe fare al caso tuo!

**Comment by shiner66** [comment]
Ho dato un'occhiata, non è niente male, ma già l'autenticazione tramite spid è un bel grattacapo da automatizzare.
Oltretutto mostra solo i dati tra una lettura e l'altra, esattamente come enel distribuzione, mi sa che di questo passo faccio prima a metter su una pinza amperometrica a valle del contatore e risolvere tutti i miei problemi :D

**Comment by trane77** [comment]
Non saprei, immagino che l'utente dovrebbe accedere al tuo servizio tramite spid, difficile forse, fattibile sicuramente, il supporto è disponibile in questa pagina https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/index.html 
La documentazione c'è, se vuoi farlo non manca niente ... la pinza amperometrica costa troppo :-D

**Comment by cagnulein** [comment]
ciao @shiner66 il portale e-distribuzione mette a disposizione i dati aggregati a 15 minuti fino al giorno prima. Stavo pensando di fare la stessa cosa (anche se non avremmo comunque i dati in tempo reale ahimè). Tu a cosa sei arrivato alla fine?

**Comment by szacchino** [comment]
> 

Ciao @cagnulein, sai se esistono endpoint rest per accedere ai propri consumi di e-distribuzione?

**Comment by cagnulein** [comment]
@szacchino io alla fine ho preso questo https://robertoviola.cloud/2022/09/03/alfa-leggi-i-tuoi-consumi-elettrici-con-un-click/
